Spigot-Server-Patches/0339-Add-LivingEntity-getTargetEntity.patch

```diff
@@ -107,18 +107,6 @@ index 7c239b95c9f8dfa76d397979e6b3fdcc9ff55022..aecb5ca746be20e31f047a68f729d5c5
      public int shieldBlockingDelay = world.paperConfig.shieldBlockingDelay;
  
      public int getShieldBlockingDelay() {
-diff --git a/src/main/java/net/minecraft/server/IEntitySelector.java b/src/main/java/net/minecraft/server/IEntitySelector.java
-index b88a99221933108141b558adef6aa91f0cedc136..c13bb708ccd14c3848afbc9afaf6439c1a574f22 100644
---- a/src/main/java/net/minecraft/server/IEntitySelector.java
-+++ b/src/main/java/net/minecraft/server/IEntitySelector.java
-@@ -18,6 +18,7 @@ public final class IEntitySelector {
-     public static final Predicate<Entity> e = (entity) -> {
-         return !(entity instanceof EntityHuman) || !entity.isSpectator() && !((EntityHuman) entity).isCreative();
-     };
-+    public static Predicate<Entity> notSpectator() { return f; } // Paper - OBFHELPER
```

Comment:
this can't be left as is.
While yes notSpectator does belong on the other method, its more so these methods were backwards

f is canAITarget and needs to be maintained, and this getTargetEntity needs to keep using this same selector as canAITarget instead of notSpectator

Reply:
I had a bit of trouble understanding what you meant here. But I hope I've fixed it correctly.

Reply:
This is still wrong, the notSpectator needs to stay on the same one it was before

return !(entity instanceof EntityHuman) || !entity.isSpectator() && !((EntityHuman) entity).isCreative() && entity.world.getDifficulty() != EnumDifficulty.PEACEFUL;

But change its name to canAITarget()

Reply:
Oh, now I think I understand it. Previously `notSpectator` was on what it's on now, but in reality in this patch, it should be used on `!(entity instanceof EntityHuman) || !entity.isSpectator() && !((EntityHuman) entity).isCreative() && entity.world.getDifficulty() != EnumDifficulty.PEACEFUL;`. And you suggest a naming change to better match the behavior.

I'll change it in a moment.